File: comment_service/nodejs/models/commentModel.js
const mongoose = require('mongoose');
const { Schema } = mongoose;

const commentSchema = new Schema({
    comment_id: { type: Schema.Types.ObjectId, required: true },
    post_id: { type: Schema.Types.ObjectId, required: true },
    user_id: { type: Schema.Types.ObjectId, required: true },
    content: { type: String, required: true },
    created_at: { type: Date, default: Date.now }
});

const Comment = mongoose

File: comment_service/python/models/commentModel.py
from pymongo import MongoClient
from bson.objectid import ObjectId
from datetime import datetime

client = MongoClient('mongodb://localhost:27017/')
db = client['comment_database']
comments = db['comments']

def create_comment(post_id, user_id, content):
    comment = {
        "comment_id": ObjectId(),
        "post_id": ObjectId(post_id),
        "user_id": ObjectId(user_id),
        "content": content,
        "created_at": datetime.utcnow()
    }
    return comments.insert_one(comment)

def get_comments_for_post(post_id):
    return list(comments.find({"post_id": ObjectId(post_id)}))
